tests/storage/restore.go

```diff
@@ -1741,7 +1743,289 @@ var _ = Describe(SIG("VirtualMachineRestore Tests", func() {
+				// Create a VM and snapshot it
+				vm, vmi = createAndStartVM(renderVMWithRegistryImportDataVolume(cd.ContainerDiskCirros, snapshotStorageClass))
+				By(creatingSnapshot)
+				snapshot = createSnapshot(vm)
+
+				const newDiskName = "new-disk"
+
+				// Create the restore definition of the VM, change the name of the restored volume
+				restoreDef := createRestoreDef(vm.Name, snapshot.Name)
+				restoreDef.Spec.TargetReadinessPolicy = ptr.To(snapshotv1.VirtualMachineRestoreStopTarget)
+				restoreDef.Spec.VolumeRestoreOverrides = []snapshotv1.VolumeRestoreOverride{
+					{
+						VolumeName:  "disk0",
+						RestoreName: newDiskName,
+						Labels:      map[string]string{"new-label": "value"},
+						Annotations: map[string]string{"new-annotation": "value"},
+					},
+				}
+
+				restore, err = virtClient.VirtualMachineRestore(vm.Namespace).Create(context.Background(), restoreDef, metav1.CreateOptions{})
+				Expect(err).ToNot(HaveOccurred())
+
+				restore = waitRestoreComplete(restore, vm.Name, &vm.UID)
+				Expect(restore.Status.Restores).To(HaveLen(1))
+
+				// Check the VM post-restore has info we want
+				restoreVM, err := virtClient.VirtualMachine(vm.Namespace).Get(context.Background(), vm.Name, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				Expect(restoreVM.Spec.Template.Spec.Volumes).To(HaveLen(1))
+				Expect(restoreVM.Spec.Template.Spec.Volumes[0].Name).To(Equal("disk0"))
+				Expect(restoreVM.Spec.Template.Spec.Volumes[0].DataVolume.Name).To(Equal(newDiskName))
+
+				// Check the restored PVC has the info we want
+				pvc, err := virtClient.CoreV1().PersistentVolumeClaims(vm.Namespace).Get(context.Background(), newDiskName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				Expect(pvc.Labels["new-label"]).To(Equal("value"))
+				Expect(pvc.Annotations["new-annotation"]).To(Equal("value"))
+
+				deleteRestore(restore)
+				restore = nil
+			})
+
+			It("should restore with volume restore policy InPlace and DV template as disk", func() {
+				// Create a VM and snapshot it
+				vm, vmi = createAndStartVM(renderVMWithRegistryImportDataVolume(cd.ContainerDiskCirros, snapshotStorageClass))
+				By(creatingSnapshot)
+				snapshot = createSnapshot(vm)
+
+				restoreDef := createRestoreDef(vm.Name, snapshot.Name)
+				restoreDef.Spec.TargetReadinessPolicy = ptr.To(snapshotv1.VirtualMachineRestoreStopTarget)
+
+				// We want to overwrite existing volumes during the restore, that means deleting the existing PVCs
+				restoreDef.Spec.VolumeRestorePolicy = ptr.To(snapshotv1.VolumeRestorePolicyInPlace)
+
+				// We're about to restore a VM in such a way that the restored volumes are identical to the source volumes.
+				// We need to make sure they're indeed new, and not that nothing happened during the test. We add
+				// a special annotation to the restored volume to ensure it has been restored from a VolumeSnapshot.
+				restoreDef.Spec.VolumeRestoreOverrides = []snapshotv1.VolumeRestoreOverride{
```

Comment:
probably enough to make sure the pvc has the restore uid annotation, but doesnt hurt :)